transaction-signer/src/main.rs

```diff
@@ -1,3 +1,378 @@
+    let account_id = request_json.get("account_id").unwrap().as_str().unwrap();
+    assert_eq!(account_secrets.account_id, account_id);
+
+    let signing_data: UnsignedTx = serde_json::from_value(
+        request_json
+            .get("unsigned_tx")
+            .expect("Could not find \"unsigned_tx\".")
+            .clone(),
+    )
+    .unwrap();
+
+    let tx = signing_data.sign(&signer, &mut rng).unwrap();
+
+    let tx_proposal = TxProposal {
+        tx,
+        input_txos: Vec::new(),
+        payload_txos: Vec::new(),
+        change_txos: Vec::new(),
+    };
+
+    let tx_proposal_json = TxProposalJSON::try_from(&tx_proposal).unwrap();
+    let json_command_request = JsonCommandRequest::submit_transaction {
+        tx_proposal: tx_proposal_json,
+        comment: None,
+        account_id: Some(account_id.to_string()),
+    };
+
+    let filename = format!(
+        "{}_completed.json",
+        sign_request.trim_end_matches("_unsigned.json")
+    );
+    write_json_command_request_to_file(&json_command_request, &filename);
+}
+
+fn write_json_command_request_to_file(json_command_request: &JsonCommandRequest, filename: &str) {
+    let src_json: serde_json::Value = serde_json::json!(json_command_request);
+    let method = src_json.get("method").unwrap().as_str().unwrap();
+    let params = src_json.get("params").unwrap();
+
+    let json_rpc_request = JsonRPCRequest {
+        method: method.to_string(),
+        params: Some(params.clone()),
+        jsonrpc: "2.0".to_string(),
+        id: serde_json::Value::Number(serde_json::Number::from(1)),
+    };
+
+    let result_json = serde_json::to_string_pretty(&json_rpc_request).unwrap();
+    fs::write(filename, result_json + "\n").expect("could not write output file");
+    println!("Wrote {}", filename);
+}
+
+fn get_key_images_for_txos(
+    tx_outs: &[TxOut],
+    account_key: &AccountKey,
+    subaddress_spend_public_keys: &HashMap<RistrettoPublic, u64>,
+) -> Vec<(TxOut, KeyImage, u64)> {
+    tx_outs
+        .iter()
+        .filter_map(|txo| {
+            if !tx_out_belongs_to_account(txo, account_key.view_private_key()) {
+                return None;
+            }
+            get_key_image_for_tx_out(txo, account_key, subaddress_spend_public_keys)
+                .map(|(key_image, subaddress_index)| (txo.clone(), key_image, subaddress_index))
+        })
+        .collect()
+}
+
+fn account_key_from_mnemonic_phrase(mnemonic_phrase: &str) -> AccountKey {
+    let mnemonic = Mnemonic::from_phrase(mnemonic_phrase, Language::English).unwrap();
+    Slip10Key::from(mnemonic)
+        .try_into_account_key("", "", &base64::decode("").unwrap())
+        .unwrap()
+}
+
+fn get_key_image_for_tx_out(
+    tx_out: &TxOut,
+    account_key: &AccountKey,
+    subaddress_spend_public_keys: &HashMap<RistrettoPublic, u64>,
+) -> Option<(KeyImage, u64)> {
+    let tx_public_key = match RistrettoPublic::try_from(&tx_out.public_key) {
```

Comment:
I think this can be simplified to `let tx_public_key = RistrettoPublic::try_from(&tx_out.public_key).ok()?;`